library/src/tests/resources/src/state_preparation.qs

```diff
@@ -106,4 +106,23 @@ namespace Test {
         }
     }
 
+    operation TestPrepareUniformSuperposition(nStates : Int) : Unit {
+        use qs = Qubit[10];
+        PrepareUniformSuperposition(nStates, qs);
+        DumpMachine();
+        ResetAll(qs);
+    }
+
+    operation TestPrepareUniformSuperpositionExhaustive() : Unit {
+        let NBits = 4;
+        use qs = Qubit[NBits];
+        for NStates in 1..2^NBits-1 {
+            PrepareUniformSuperposition(NStates, qs);
+            let t = Std.Math.Sqrt(1.0 / IntAsDouble(NStates));
+            Adjoint Std.StatePreparation.PreparePureStateD(Repeated(t, NStates), Reversed(qs));
+            Fact(CheckAllZero(qs), "All qubits must be in zero state.");
+            ResetAll(qs);
```

Comment:
nitpick: since you have a `Fact` above that the qubits are all zero there is no need to reset them here.
```suggestion
```

Reply:
Removed.